src/hsmm_mvpy/visu.py

```diff
@@ -86,78 +86,94 @@ def plot_topo_timecourse(channels, estimated, channel_position, init, time_step=
     from mne.viz import plot_brain_colorbar, plot_topomap
     from mne.io.meas_info import Info
     from mpl_toolkits.axes_grid1.inset_locator import inset_axes
-    from mpl_toolkits.axes_grid1 import make_axes_locatable
+    #from mpl_toolkits.axes_grid1 import make_axes_locatable
 
     return_ax = True
+    #if not times specified, plot average RT
     if times_to_display is None:
         times_to_display = init.mean_d
+    
+    #set xlabel depending on time_step
     if xlabel is None:
         if time_step == 1:
             xlabel = 'Time (in samples)'
         else:
             xlabel = 'Time'
+
+    #set color of event_lines 
     if event_lines == True:
         event_color='tab:orange'
     else:
         event_color=event_lines
 
-    
+    #if estimated is an fitted HMP instance
     if isinstance(estimated, (xr.DataArray, xr.Dataset)) and 'event' in estimated:
-        if ydim is None and 'n_events' in estimated.dims:
-            if estimated.n_events.count() > 1:
-                ydim = 'n_events'
-        if ydim is not None and not skip_channels_computation:
-            channels = init.compute_topologies(channels, estimated, init.event_width_samples, ydim).data
-        elif not skip_channels_computation:
-            channels = init.compute_topologies(channels, estimated, init.event_width_samples).data
-        times = init.compute_times(init, estimated, mean=True).data
-        channels[times == 0] = np.nan#removes the empty topologies, e.g. in the case of mutliple varying number of events
-
-    else:#assumes times already computed
+        if ydim is None and 'n_events' in estimated.dims: #and there are multiple different fits (eg backward estimation)
+            if estimated.n_events.count() > 1: #and we really have more than a singel fit
+                ydim = 'n_events' #set ydim to 'n_events'
+        if not skip_channels_computation:
+            channels = init.compute_topologies(channels, estimated, init.event_width_samples, ydim).data #compute topologies
```

Comment:
We lost a condition here right? Doesn't it impact single fit plotting?

Reply:
This also works because ydim will be `None`, which evaluates to False apparently:
<img width="154" alt="Screenshot 2023-09-01 at 14 08 50" src="https://github.com/GWeindel/hsmm_mvpy/assets/19282335/e2d7f79f-5b0a-45db-a2c6-fa8e0ce03b4a">

Reply:
OK then it's all good